app/components/global_threads/global_threads.tsx

```diff
@@ -108,6 +115,23 @@ function GlobalThreadsList({actions, allThreadIds, intl, teamId, theme, threadCo
         );
     };
 
+    const goToThread = React.useCallback((post: Post) => {
+        actions.getPostThread(post.id);
+        actions.selectPost(post.id);
+        const passProps = {
+            channelId: post.channel_id,
+            rootId: post.id,
+        };
+        goToScreen(THREAD, '', passProps);
+    }, []);
+
+    React.useEffect(() => {
+        EventEmitter.on('goToThread', goToThread);
+        return () => {
+            EventEmitter.off('goToThread', goToThread);
+        };
+    }, []);
```

Comment:
how is this different from `channel_post_list`? is channel_post_list not mounted when this component is on screen?

Reply:
No it's not getting mounted. We are already using this code for navigating to the thread on pressing on the list item .. now we are also using it for the menu item.